test/graphql-api/utils/executeOperation.ts

```diff
@@ -51,21 +53,45 @@ export const executeOperation = async (
 
   const urlString = url.toString()
 
+  // Cannot pass files because of Puppeteer's limitation.
   const responsePromise = page.evaluate(
-    (url, method, query, variables) => {
+    (
+      url,
+      method,
+      query,
+      variables,
+      map,
+      fileContents: string[] | undefined,
+    ) => {
+      const operations = JSON.stringify({
+        query,
+        variables,
+      })
+
+      const isMultipart = Boolean(map && fileContents)
+      const headers = isMultipart ? {} : { 'Content-Type': 'application/json' }
+      let body: string | FormData
```

Comment:
I'd refactor this part as follows:

```js
const body = isMultipartData ? getMultipartGraphQLBody(operations, ...) : operations
```

This will make `executeOperations` much easier to read by delegating the logic that creates a `FormData` into a separate utility function that we can also test separately.